src/System.Text.Utf8/src/System/Text/Utf8/Utf8CodeUnit.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace System.Text.Utf8
+{
+    public struct Utf8CodeUnit : IEquatable<Utf8CodeUnit>
```

Comment:
I think public types need to add enough value to justify their existence. I don't think Utf8CodeUnit adds enough value. And we are not consistent with System.String anyway. Utf8String is actually a sub-string (slice of string), it's not immutable, and amny other differences.